kinescope/kinescope.py

```diff
@@ -0,0 +1,110 @@
+"""This XBlock embeds content from Kinescope through Iframes"""
+
+import pkg_resources
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope, String
+try:
+    from xblock.utils.studio_editable import StudioEditableXBlockMixin
+except ModuleNotFoundError: # For compatibility with Palm and earlier
+    from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblock.validation import ValidationMessage
+
+
+from .utils import _, is_url, parse_valid_kinescope_url
+
+
+class KinescopeXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    This XBlock renders Iframe for Kinescope videos.
+    """
+
+    display_name = String(
+        display_name=_("Display name"),
+        default=_("Kinescope"),
+        scope=Scope.settings,
+        help=_("Display name for this XBlock."),
+    )
+
+    video_link = String(
+        display_name="Video Link/URL",
+        default="",
+        scope=Scope.content,
+        help=_(
+            "Video link copied from Kinescope dashboard. The video id below is extracted "
+            "from this link if provided"
+        )
+    )
+
+    video_id = String(
+        display_name=_("Video ID"),
+        default="",
+        scope=Scope.content,
+        help=_(
+            "UUID of the video to embed."
```

Comment:
I agree that `ID` would be more appropriate here than `UUID`.

I added `UUID` because of [this screen-grab](https://file.notion.so/f/s/3f740031-128d-4709-98c3-fa8ed82b1621/Screen_Recording_2023-03-15_at_13.29.56.gif?id=5bdff74b-2f2c-4e07-8cd7-00e1afcc4d11&table=block&spaceId=9478f27e-ac27-4b22-9047-80233ecc5351&expirationTimestamp=1698739200000&signature=gPtUB2ND5oSrahqb5EwVle-9X7tWezqDqQbMyImZWXE&downloadName=Screen+Recording+2023-03-15+at+13.29.56.gif) from the same page shows the option of `Copy video UUID`.

In anycase, this field is not visible/editable anymore as I have explained in a comment below.